core/controllers/base_test.py

```diff
@@ -87,6 +87,18 @@ def test_requests_for_invalid_paths(self):
         response = self.testapp.put('/gallery/extra', {}, expect_errors=True)
         self.assertEqual(response.status_int, 404)
 
+    def test_redirect_in_both_logged_in_and_logged_out_states(self):
+        "Test for a redirect in both logged in and logged out states on '/'."
+
+        # Logged out state
+        response = self.testapp.get('/')
+        self.assertEqual(response.status_int, 302)
```

Comment:
@seanlip :+1:Thank you, for not just finding a solution to this, but also for explaining your debug process.

I'm really getting hold of a lot about the development workflow and the testing process followed at Oppia, thanks to your comments.

Also, yes it worked when I registered the user first.